### DISCUSSION TITLE
[@types/node] Is it still maintained? 🙄

### DISCUSSION THREAD

**Discussion by vsDizzy** [DefinitelyTyped/DefinitelyTyped#69754] [Issues with a @types package] [answered]
I noticed there are no typings for the features that are already in the Node v22
- [Array.fromAsync()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fromAsync#browser_compatibility)
- [Iterator.prototype.take()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/take#browser_compatibility)

[a comment by an automated account is not shown]

**Reply by ghost** [reply, marked as answer]
It is still maintained, but this is not something exclusive to node v22. This is related to DOM and the typings need to updated

**Reply by vincerubinetti** [reply]
I found this looking for any open or closed issues or PRs for Node v22. It was just released at the end of April, but I didn't realize it won't be in "LTS" until October. Is there a policy in the `@types` repo to not add types for it until something is a "stable" release? Or is the org just a little behind on adding the types (I understand that it's a big effort, that there are a lot of new APIs and such).

In the meantime, does anyone know if there is a type "shim" or work-in-progress Node v22 typing somewhere, so I can use the [new `Set` features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/symmetricDifference)?

**Reply by vsDizzy** [reply]
Array.fromAsync is already in the typescript typings https://github.com/microsoft/TypeScript/blob/6f06eb1b27a6495b209e8be79036f3b2ea92cd0b/src/lib/esnext.array.d.ts#L6

**Reply by vincerubinetti** [reply]
Ah right I forgot that regular browser/DOM API type definitions like this are builtin to TS. For APIs that are supposed to be the same between the browser and Node, does `@types/node` just not have type definitions? Does it just import them from TS and re-export them? Does it duplicate them?

Anyway, it looks like my desired `Set` type defs are here: 
https://github.com/microsoft/TypeScript/commit/e8274f74f9f94d2fc9677023b27746888e288b59
https://github.com/microsoft/TypeScript/releases/tag/v5.5-rc

Installed `v5.5` (or higher) and had to manually `import "typescript/lib/lib.esnext.collection.d.ts"`.

**Reply by vsDizzy** [reply]
> Installed `v5.5` (or higher) and had to manually `import "typescript/lib/lib.esnext.collection"`.

I use next version:
`npm i -D typescript@next`

And instead of import you can just add lib to `tsconfig.json`: ` "lib": ["ESNext"]`
E.g.
```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "rootDir": "src",
    "lib": ["ESNext"],
    "skipLibCheck": true,
    "outDir": "dist"
  }
}
```

**Reply by vincerubinetti** [reply]
I already had that in my `tsconfig`. TS was still complaining about the `Set.difference` method not being present.

```json
{
  "compilerOptions": {
    "esModuleInterop": true,
    "isolatedModules": true,
    "lib": ["ESNext"],
    "module": "ESNext",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noErrorTruncation": true,
    "noUncheckedIndexedAccess": true,
    "paths": {
      "@/*": ["./*"]
    },
    "resolveJsonModule": true
  },
  "extends": "@tsconfig/node22/tsconfig.json",
  "include": ["*"]
}
```

**Reply by vsDizzy** [reply]
For me it is working:

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5798340/586f8325-0ad3-49f0-9442-c398eb35f788)

**Reply by Anonymous4078** [reply]
> I found this looking for any open or closed issues or PRs for Node v22. It was just released at the end of April, but I didn't realize it won't be in "LTS" until October. Is there a policy in the `@types` repo to not add types for it until something is a "stable" release? Or is the org just a little behind on adding the types (I understand that it's a big effort, that there are a lot of new APIs and such).
> 
> In the meantime, does anyone know if there is a type "shim" or work-in-progress Node v22 typing somewhere, so I can use the [new `Set` features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/symmetricDifference)?

It was being maintained by someone named codershiba who has now left GitHub so it's upto the other community members or maintainers.
